packages/block-library/src/media-text/edit.js

```diff
@@ -145,9 +150,42 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 		rel,
 		verticalAlignment,
 		allowedBlocks,
+		useFeaturedImage,
 	} = attributes;
 	const mediaSizeSlug = attributes.mediaSizeSlug || DEFAULT_MEDIA_SIZE_SLUG;
 
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const featuredImageMedia = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+
+	const featuredImageURL = useFeaturedImage
+		? featuredImageMedia?.source_url
+		: '';
+	const featuredImageAlt = useFeaturedImage
+		? featuredImageMedia?.alt_text
+		: '';
```

Comment:
Since we check for `featuredImageURL` and `featuredImageAlt` as `bool` in many places can we set them to `false` if they have no value? IMO the way JS says ` false || ''` is `false`  is better if we default to testing for `false`? Is there any downside?

Alternatively we could make `hasFeaturedImageAlt` and `hasFeaturedImageURL`, respectively. set to ` !== '' `, then use these in `bool` checks later?

Reply:
I tried this, but it meant that the alt text was set to the string "false" instead of being empty when:

- The featured image is not used.
- The current media has no alt text of its own.

It shows the text "false" inside the alt text option in the editor, and in the attribute.

[So this condition on line 316 also needs to be updated.
`value={ mediaAlt || featuredImageAlt }`](https://github.com/WordPress/gutenberg/pull/51491/files#diff-799bc423023028d1e733982e9d106233470b262e34737b34c559ab7b14e001d7R316)

Reply:
@draganescu I think i did not fully understand what to do here.

Reply:
> It shows the text "false" inside the alt text option in the editor, and in the attribute.

It means we need the empty string, In which case it means we should not check for `value={ mediaAlt || featuredImageAlt }` later but instead for `value={ mediaAlt || featuredImageAlt !== '' }`. That's it :)